### ISSUE TITLE
Could you provide pretrained model of discriminator

### ISSUE THREAD

**Issue by tututu05** [chaofengc/PSFRGAN#9]
It's a very great job. I want to reproduce your work and finetune in my dataset. Could you provide pretrained model of discriminator if possible?
Thank you very much!

**Comment by chaofengc** [comment]
I am sorry that I abandoned the discriminator weight of the pretrained generator. The structure of discriminator is in `models/networks`, you may use the pretrained generator with a random initialized discriminator to finetune the model on your data. According to my experience, this takes a longer time to train (results may become worse at the begining) but does not have much influence to the final performance.

**Comment by tututu05** [comment]
> I am sorry that I abandoned the discriminator weight of the pretrained generator. The structure of discriminator is in `models/networks`, you may use the pretrained generator with a random initialized discriminator to finetune the model on your data. According to my experience, this takes a longer time to train (results may become worse at the begining) but does not have much influence to the final performance.

when I read your code in ehance_model.py, I found that the loss function in the code is different from that in the paper. So which one is right?

**Comment by chaofengc** [comment]
Which loss do you mean ?

**Comment by tututu05** [comment]
the perceptual loss and L1 loss is used in your code but MSE in your paper

**Comment by chaofengc** [comment]
Sorry, that's my mistake. For this case, L1 and L2 loss does not make much visual difference according to my experience. To get the same results as the provided model, you should follow the codes.

**Comment by tututu05** [comment]
Thank you for your reply

**Comment by tututu05** [comment]
> Sorry, that's my mistake. For this case, L1 and L2 loss does not make much visual difference according to my experience. To get the same results as the provided model, you should follow the codes.

I tried to train psfrgan with my reproduced training code . The dataset is FFHQ, and other parameters was not changed. But the result is weired, espically the texture of hair and face is not clear. Can you give me some some advice?

**Comment by chaofengc** [comment]
I am not sure what is your problem. You'd better provide the examples.
Besides, it takes a lot of time (about 3 days with batch size of 4) to get good results.

**Comment by tututu05** [comment]
> I am not sure what is your problem. You'd better provide the examples.
> Besides, it takes a lot of time (about 3 days with batch size of 4) to get good results.

Thank you for your reply. It's test result of model that I trained:
![20201105231010](https://user-images.githubusercontent.com/43262795/98469758-75759180-221c-11eb-80ba-7230e7efd199.jpg)
I use all images of FFHQ to train, and trained the model about 20 epochs.  Unfortunately, I can't get your result.

**Comment by chaofengc** [comment]
You should follow the degradation model provided in the paper to generate the LQ images.
It seems that your model can not remove the jpeg artifacts. I guess you did not include the JPEG compression in your degradation model.

**Comment by tututu05** [comment]
My degradation model have JPEG compression. The biggest problem with the trained model is that it can't remove the blur and noise of the test picture and can't generate clear textures. which epoch did you stop training at?

**Comment by chaofengc** [comment]
In my experience, 20 epochs training of FFHQ is more than enough. First of all, **you should check your model results during the training process**, because the training results should be quite good with clear textures regardless of the degradation model you use.

If the training results are good, but testing results are bad. The problem should lie in the degradation model you implement. Please follow exactly the degradation equation in the paper, make sure you did not change the order, to be specific, use the following steps:
1. Blur the image with blur kernel randomly chosen from: gaussian blur, motion blur, average blur and median blur
1. Downsample the image
1. Add noise
1. Do JPEG compression

I use [imgaug](https://imgaug.readthedocs.io/en/latest/) to implement the degradation model. The JPEG should be done at the final step, otherwise the model may not be able to generate good results for real world images.

There is no other tricks in the training except for the training details in the paper. As for the degradation model parameters, I just choose some figures intuitively without further experiments. You should finetune the degradation parameters to achieve the best performance for your own data. 

**This is the end for this issue.** I cannot help you more if you still can not get good results. Please wait for the release of all codes.

**Comment by tututu05** [comment]
Thank you for your help. I will try again

**Comment by erjihaoshi** [comment]
@tututu05    do you implement the train code of psfrgan?  can we chat in privately?
